build/publish/Publish.csproj

```diff
@@ -0,0 +1,186 @@
+        <BlobBasePath>aspnetcore/npm/$(PackageVersion)/</BlobBasePath>
+      </NpmPackageToPublish>
+      <_FilesToPublish Include="@(NpmPackageToPublish)" />
+
+      <!--
+        Transform the intermediate item group into the final group.
+        You can't use globbing _and_ set metadata using FileName and Extension at the same time. MSBuild quirks are fun.
+      -->
+      <FilesToPublish Include="@(_FilesToPublish)">
+        <RelativeBlobPath Condition="'%(_FilesToPublish.BlobBasePath)' != ''">%(_FilesToPublish.BlobBasePath)%(_FilesToPublish.FileName)%(_FilesToPublish.Extension)</RelativeBlobPath>
+        <RelativeBlobPath Condition="'%(_FilesToPublish.BlobBasePath)' == ''">aspnetcore/Runtime/$(PackageVersion)/%(_FilesToPublish.FileName)%(_FilesToPublish.Extension)</RelativeBlobPath>
+      </FilesToPublish>
+      <_FilesToPublish Remove="@(_FilesToPublish)" />
+
+      <PackageToPublish Include="$(ProductPackageOutputPath)*.symbols.nupkg" IsSymbolsPackage="true" />
+      <PackageToPublish Include="$(ProductPackageOutputPath)*.nupkg" Exclude="@(PackageToPublish)" />
+    </ItemGroup>
+
+    <!--
+      'Internal' packages are used to transfer bits to partner teams, and should not be used by customers.
+      Publishing these can be disabled.
+    -->
+    <ItemGroup Condition=" '$(PublishInternalPackages)' != 'false' ">
+      <PackageToPublish Include="$(InternalPackageOutputPath)*.symbols.nupkg" IsSymbolsPackage="true">
+        <ManifestArtifactData>NonShipping=true</ManifestArtifactData>
+      </PackageToPublish>
+      <PackageToPublish Include="$(InternalPackageOutputPath)*.nupkg" Exclude="@(PackageToPublish)">
+        <ManifestArtifactData>NonShipping=true</ManifestArtifactData>
+      </PackageToPublish>
+    </ItemGroup>
+
+    <Error Text="Missing required metadata 'RelativeBlobPath' for FilesToPublish: @(FilesToPublish)" Condition="'%(RelativeBlobPath)' == ''" />
+  </Target>
+
+  <UsingTask TaskName="KoreBuild.Tasks.PushNuGetPackages" AssemblyFile="$(KoreBuildTasksDll)" />
+
+  <Target Name="PublishToMyGet"
+          DependsOnTargets="GetFilesToPublish"
+          Condition="'$(PublishToMyget)' == 'true'">
+
+    <Error Text="Missing required property: PublishMyGetFeedUrl"  Condition=" '$(PublishMyGetFeedUrl)' == '' "/>
+    <Error Text="Missing required property: PublishMyGetSymbolsFeedUrl"  Condition=" '$(PublishMyGetSymbolsFeedUrl)' == '' "/>
+    <Error Text="Missing required property: PublishMyGetNpmRegistryUrl"  Condition=" '$(PublishMyGetNpmRegistryUrl)' == '' "/>
+    <Error Text="Missing required property: PublishMyGetFeedKey"  Condition=" '$(PublishMyGetFeedKey)' == '' "/>
+
+    <Error Text="No packages found to publish" Condition="@(PackageToPublish->Count()) == 0" />
+
+    <PushNuGetPackages Condition="'%(PackageToPublish.IsSymbolsPackage)' != 'true' AND @(PackageToPublish->Count()) != 0"
+      Packages="@(PackageToPublish)"
+      TimeoutSeconds="300"
```

Comment:
Nice. In some places we use as high as 600 seconds, which is what I was planning to change everything to "for consistency." Up to you, but I might go raise it anyway when I try to de-flakify the NuGet pushes.